File: benchmark/ruby/palindrome.rb
s = "a" * 50000

def is_pal(str)
    left = 0
    right = str.length - 1
    while left < right
        if str[left] != str[right]
            return false
        end
        left += 1
        right -= 1
    end
    true
end

start = Time.now.to_f * 1000
ok = true
for i in 0...1000
    ok = is_pal(s)
end
finish = Time.now.to_f * 1000
puts "Palindrome: #{ok}"
puts "Time: #{(finish - start).round}ms"

File: benchmark/nodejs/palindrome.js
const s = "a".repeat(50000);

function is_pal(str) {
    let left = 0;
    let right = str.length - 1;
    while (left < right) {
        if (str[left] !== str[right]) {
            return false;
        }
        left++;
        right--;
    }
    return true;
}

const start = Date.now();
let ok = true;
for (let i = 0; i < 1000; i++) {
    ok = is_pal(s);
}
const end = Date.now();
console.log(`Palindrome: ${ok}`);
console.log(`Time: ${end - start}ms`);

